app/services/request_password_reset.rb

```diff
@@ -1,16 +1,15 @@
 RequestPasswordReset = Struct.new(:email) do
   def perform
-    if user_not_found? || user_found_but_is_an_admin_or_tech?
-      # do nothing
-      #
-      # To prevent revealing account existence, we don't treat a
-      # "user not found" scenario as an error, unlike Devise's default
-      # behavior. That way, when the PasswordsController finishes processing
-      # the action, it will look like a successful transaction. We act as if
-      # the password reset email was sent, when in fact it wasn't.
-      # Similary, for security purposes, we don't want to allow password
-      # recovery via email for admin and tech support users.
-    elsif user.confirmed?
+    # To prevent revealing account existence, we don't treat a
+    # "user not found" scenario as an error, unlike Devise's default
+    # behavior. That way, when the PasswordsController finishes processing
+    # the action, it will look like a successful transaction. We act as if
+    # the password reset email was sent, when in fact it wasn't.
+    # Similary, for security purposes, we don't want to allow password
+    # recovery via email for admin and tech support users.
+    return if user_not_found? || user_found_but_is_an_admin_or_tech?
```

Comment:
Personally, I like early returns because it's this growing 'guard' style of programming which Swift now uses; also I like the less-deeply nested structures which result.